test/extensions/filters/http/oauth2/filter_test.cc

```diff
@@ -155,6 +157,82 @@ class OAuth2Test : public testing::Test {
   Event::SimulatedTimeSystem test_time_;
 };
 
+// Verifies that the OAuth SDSSecretReader correctly updates dynamic generic secret
+TEST_F(OAuth2Test, SdsDynamicGenericSecret) {
+  NiceMock<Server::MockConfigTracker> config_tracker;
+  std::unique_ptr<Secret::SecretManager> secret_manager(
+      new Secret::SecretManagerImpl(config_tracker));
+  envoy::config::core::v3::ConfigSource config_source;
+
+  NiceMock<Server::Configuration::MockTransportSocketFactoryContext> secret_context;
+  NiceMock<LocalInfo::MockLocalInfo> local_info;
+  Api::ApiPtr api = Api::createApiForTest();
+  Stats::IsolatedStoreImpl stats;
```

Comment:
At least this one can use the store_ defined on the fixture class instead of redefining one here

Reply:
I think it will be more readable if we leave it in the function scope to save the context.